pkg/search/postgres/store_cache.go

```diff
@@ -225,20 +225,14 @@ func (c *cachedStore[T, PT]) Exists(ctx context.Context, id string) (bool, error
 	return c.isReadAllowed(ctx, obj), nil
 }
 
-// Count returns the number of objects in the store.
-func (c *cachedStore[T, PT]) Count(ctx context.Context) (int, error) {
-	defer c.setCacheOperationDurationTime(time.Now(), ops.Count)
-	c.cacheLock.RLock()
-	defer c.cacheLock.RUnlock()
-	count := 0
-	err := c.walkCacheNoLock(ctx, func(obj PT) error {
-		count++
-		return nil
-	})
-	if err != nil {
-		return 0, err
-	}
-	return count, nil
+// Count returns the number of objects in the store matching the query.
+func (c *cachedStore[T, PT]) Count(ctx context.Context, q *v1.Query) (int, error) {
+	return c.underlyingStore.Count(ctx, q)
```

Comment:
This is now much slower because it will always hit the underlying store and not the cache

Reply:
When I first looked at merging the indexed data searcher in the stores, I thought about splitting the Count function of the store from the searcher one. The Store one is not used so much, and in most cases it is just for tests.
If there are performance issues, then we may want to have this function split. I fear implementing Count by query on cached store would be equivalent to reverting to the KV+Indexer architecture we had before moving to an external relational DB.

Reply:
I think we retain the cache lookup for certain cases.

Reply:
I don't think this conversation is resolved. Count now always hits the underlying store which will result in a postgres query where previously it would have been resolved via a walk of the cache which would most likely be much more performant

Reply:
I think I disagree. The cache is looked up if there is no query or if the query is empty, which was behavior of Count function before this change.

Reply:
Chose to go with `proto.Equal` since it is safer when `EmptyQuery()` abstracts a different proto object.